### PR TITLE
[AppBar] Fix WindowInsets handling with AppBarLayout

### PR DESCRIPTION
**Pull request by kfaraj**
### Thanks for starting a pull request on Material Components!

Closes https://github.com/material-components/material-components-android/issues/2120

[Contributing](https://github.com/material-components/material-components-android/blob/master/docs/contributing.md)
has more information and tips for a great pull request.

### CODE REVIEW

**Comment by b95505017** [comment]
@wcshi Any change merge this PR?

File: lib/java/com/google/android/material/appbar/HeaderScrollingViewBehavior.java [hunk 1]
```diff
@@ -72,8 +72,7 @@ public boolean onMeasureChild(
           if (ViewCompat.getFitsSystemWindows(header)) {
             final WindowInsetsCompat parentInsets = parent.getLastWindowInsets();
             if (parentInsets != null) {
-              availableHeight += parentInsets.getSystemWindowInsetTop()
-                  + parentInsets.getSystemWindowInsetBottom();
+              availableHeight += parentInsets.getSystemWindowInsetTop();
```
Comment (drchen) [inline lib/java/com/google/android/material/appbar/HeaderScrollingViewBehavior.java:75]: Can you provide more details in the commit message about why this fixes the issue and what will be the impact if we count bottom insets into available heights? Some screenshots will be even better. Thanks!

**Comment by drchen** [comment]
@dsn5ft can you take a look?

File: lib/java/com/google/android/material/appbar/HeaderScrollingViewBehavior.java (hunk 1 above)
Comment (kfaraj) [reply lib/java/com/google/android/material/appbar/HeaderScrollingViewBehavior.java:75]: **Description:** The available height for the scrolling content includes the bottom inset. As a consequence, the scrolling content is laid out behind the keyboard.

**Expected behavior:** The available height for the scrolling content does not include the bottom inset. As a consequence, the scrolling content is laid out above the keyboard.

**Comment by kfaraj** [comment]
As suggested, I provided more details in the commit message and added some screenshots.

```
<?xml version="1.0" encoding="utf-8"?>
<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:fitsSystemWindows="true">

    <com.google.android.material.appbar.AppBarLayout
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:fitsSystemWindows="true">

        <com.google.android.material.appbar.MaterialToolbar
            android:id="@+id/toolbar"
            android:layout_width="match_parent"
            android:layout_height="?attr/actionBarSize" />

    </com.google.android.material.appbar.AppBarLayout>

    <androidx.fragment.app.FragmentContainerView
        android:id="@+id/container"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        app:layout_behavior="@string/appbar_scrolling_view_behavior" />

</androidx.coordinatorlayout.widget.CoordinatorLayout>
```

Screenshot_20221118_152521

Without the fix, the available height for the scrolling content includes the bottom inset and the scrolling content is laid out behind the keyboard.

Screenshot_20221118_152553

With the fix, the available height for the scrolling content does not include the bottom inset and the scrolling content is laid out above the keyboard.

Screenshot_20221118_152759

**Comment by dsn5ft** [comment]
I'm really not sure that we can just remove the `parentInsets.getSystemWindowInsetBottom()`. I didn't implement the original component, but I suspect it might have been added because of translucent / transparent system navigation bars, which you might want the scrolling content to be underneath (as opposed to the keyboard).

**Comment by dsn5ft** [comment]
Maybe the answer is to use `insets.getInsets(WindowInsetsCompat.Type.systemBars()).top` and `insets.getInsets(WindowInsetsCompat.Type.systemBars()).bottom` here instead of `getSystemWindowInsetTop()` / `getSystemWindowInsetBottom()` which are deprecated.

The newer `WindowInsetsCompat.Type#systemBars()` excludes IME (keyboard):

https://developer.android.com/reference/androidx/core/view/WindowInsetsCompat.Type#systemBars()

**Pull request closed without merging**